src/vs/editor/browser/viewParts/lineNumbers/lineNumbers.ts

```diff
@@ -103,22 +118,31 @@ export class LineNumbersOverlay extends DynamicViewOverlay {
 		let visibleStartLineNumber = ctx.visibleRange.startLineNumber;
 		let visibleEndLineNumber = ctx.visibleRange.endLineNumber;
 		let common = '<div class="' + ClassNames.LINE_NUMBERS + lineHeightClassName + '" style="left:' + this._lineNumbersLeft.toString() + 'px;width:' + this._lineNumbersWidth.toString() + 'px;height:' + lineHeight + 'px;">';
+		let commonLeft = '<div class="' + ClassNames.LINE_NUMBERS + lineHeightClassName + '" style="text-align:left;left:' + this._lineNumbersLeft.toString() + 'px;width:' + this._lineNumbersWidth.toString() + 'px;height:' + lineHeight + 'px;">';
 
 		let output: string[] = [];
 		for (let lineNumber = visibleStartLineNumber; lineNumber <= visibleEndLineNumber; lineNumber++) {
 			let lineIndex = lineNumber - visibleStartLineNumber;
+			let position = this._context.model.convertViewPositionToModelPosition(lineNumber, 1);
+			let relativeLineNumber = position.lineNumber;
+
+			if (position.column === 1) {
+				if (this._relativeLineNumbers) {
```

Comment:
The test `if (this._relativeLineNumbers)` should be the first thing done here before doing any position conversions. All the new rendering code should go in this `if`. All the old rendering code should be maintained in an `else` branch. As it looks right now, regular line number rendering appears to be broken.